### PR TITLE
Ajax filter

### LINKED ISSUE #154: Deal with filters that have commas in their names
Filters taht have commas in their names, eg. locations are not working eith the filtering

### PR DESCRIPTION
**Pull request by Danwhy**
refs issue #146 

Makes filters work client side using page.js/fetch.

Also fixes #154

### CODE REVIEW

**Comment by SimonLab** [comment]
LGTM, @nelsonic?

File: client/lib/split-commas.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module.exports = function splitOnUnescapedCommas (str) {
```
Comment (nelsonic) [inline client/lib/split-commas.js]: this file deserves a `JSDoc` _comment_ explaining its' purpose ...

File: lib/facets/aggs-people.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  }
+
+  return filters.map(el => {
+    var filt = {
+      terms: {}
+    };
+    if (el.terms && Array.isArray(el.terms[Object.keys(el.terms)[0]])) {
+      filt.terms[Object.keys(el.terms)[0]] = el.terms[Object.keys(el.terms)[0]].map(el => {
+        return el.replace(/\\,/g, ',');
```
Comment (nelsonic) [inline lib/facets/aggs-people.js]: worth leaving a _comment_ to _clarify_ why you're replacing `\\` with `,` ...

File: lib/facets/create-filter-objects.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    filters.push({ terms: { 'lifecycle.creation.places.name.value': places } });
+  }
+
+  // user
+  const user = queryParams.filter.objects.user;
+  if (user) {
+    filters.push({ terms: { 'agents.summary_title': user } });
+  }
+
+  return filters.map(el => {
```
Comment (nelsonic) [inline lib/facets/create-filter-objects.js]: this _looks_ familiar... (_duplicate code..._)

File: lib/facets/create-filter-people.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  const deathDate = queryParams.filter.people.deathDate;
+  if (Object.prototype.toString.call(deathDate) === '[object Date]') {
+    if (!isNaN(deathDate.getTime())) {
+      filters.push({ range: { 'lifecycle.death.date.latest': { 'lte': deathDate.getFullYear() } } });
+    }
+  }
+
+  return filters.map(el => {
```
Comment (nelsonic) [inline lib/facets/create-filter-people.js]: ok. we _definitely_ need to DRY this out. (_please add task to backlog_)

File: lib/transforms/search-results-to-jsonapi.js [hunk 5]
```diff
@@ -260,6 +260,137 @@ function createCountCategories (results) {
   return result;
 }
 
+/**
+* Create the facets object depending of the type of hte search (all, people, objects, documents)
+*/
+function createFacets (type, results) {
```
Comment (nelsonic) [inline lib/transforms/search-results-to-jsonapi.js]: See: https://github.com/TheScienceMuseum/collectionsonline/pull/140#discussion_r70432229

File: test/client/filters.clienttest.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
+module.exports = {
+  'Filtering on search page': function (browser) {
+    browser
+      .url('http://localhost:8000/search?q=charles')
```
Comment (nelsonic) [inline test/client/filters.clienttest.js]: this Nightwatch test looks *good*. 
are you running it from CI?
you don't _appear_ to be saving any screenshots of the output, might be a good addition. 
did you see: https://github.com/dwyl/learn-nightwatch/pull/27 ?

Comment (Danwhy) [inline test/client/filters.clienttest.js]: Yeah we're running it on travis via saucelabs, so we can see the videos there. I do want to do a bit more with nightwatch this week as part of #162, so I'll probably save some screenshots and add more browsers

**Comment by nelsonic** [comment]
@Danwhy / @SimonLab code looks good. thanks for addressing comments. 👍

File: test/client/filters.clienttest.js (hunk 6 above)
Comment (nelsonic) [inline test/client/filters.clienttest.js]: I find that running the tests on Saucelabs is good for _pass/fail_ but no non-technical person is going to login to sauce, find the test and go through the videos/screenshots ... we need to make it *easy* ...

**Pull request merged**